internal/output/group.go

```diff
@@ -17,7 +18,17 @@ func (g Group) WrapWriter(stdOut, _ io.Writer, _ string, tmpl Templater) (io.Wri
 	if g.End != "" {
 		gw.end = tmpl.Replace(g.End) + "\n"
 	}
-	return gw, gw, func() error { return gw.close() }
+	return gw, gw, func(errored bool) error {
+		if g.shouldSwallowOutput(errored) {
```

Comment:
1. I have the impression that passing an error would better follow Go's conventions (no strong opinion, though)
2. In theory, instead of clearing the buffer we could just... do nothing

```go
	return gw, gw, func(err error) error {
		if g.ErrorOnly && err == nil {
			return nil
		}

		return gw.close()
```

Reply:
done